File: Program7/FileConversion.py
"""
Jeffrey Lansford
11/2/2020
Program 7
Python version of removing content from file when seeing a Control-C and stopping when seeing a Control-B
"""

import sys

if __name__ == "__main__":

    # command line arguemnts for file input and output
    fileInputName = sys.argv[1]
    fileOutputName = sys.argv[2]

    # set to to write input to output file
    deletemode = False
    # open output file for writing
    fileOutput = open(fileOutputName, "w")
    # open corrupted file
    with open(fileInputName, "r", encoding="ascii") as fileInput:
        while True:
            # read in one character at a time
            c = fileInput.read(1)
            # reached End of File
            if not c:
                print("End of File")
                break
            # see Control-C character acsii value and start not writing input to output file
            if ord(c) == 3:
                deletemode = True

            # write input to output if we are not in delete mode
            if not deletemode:
                fileOutput.write(c)
            # when seeing a Control-B character ascii value, start writing to ouptut file
            if ord(c) == 2:
                deletemode = False

        # close files
        fileInput.close()
        fileOutput.close()
        print(f"Successfully wrote to {fileOutputName}")

File: Program7/FileConversion.pl
#!/usr/bin/perl
# Jeffrey Lansford
# 11/2/2020
# Program 7
# Perl version of removing content from file when seeing a Control-C and stopping when seeing a Control-B

use strict;
use warnings;

# file name arguments for in and out
my ($fileInputName,$fileOutputName) = @ARGV;

# open  Input file 
open(INPUT, "<".$fileInputName) or die "Couldnt open file, $!";

# open Output file
open(OUTPUT, ">".$fileOutputName) or die "Couldnt open file, $!";

# set to writing mode
my $deleteMode = 0;
while(1) {
    # read a charcter at a time  
    my $char;
    read INPUT, $char, 1;

    # reached end of file
    if (!ord($char)) {
        last;
    }

    # start delete mode when seeing a Control-C character ascii value
    if (ord($char) == 3) {
        $deleteMode = 1;
    } 

    # write to oupt when in write mode
    if (!$deleteMode) {
        print OUTPUT $char;
    }

    # switch to write mode when seeing a Control-B character ascii value
    if (ord($char) == 2) {
        $deleteMode = 0;
    }

}

# close files
close(INPUT) || die "Couldn't close file properly";
close(OUTPUT) || die "Couldn't close file properly";

print "Successfully wrote to control-char-output-Perl.txt\n";

